src/main/scala/PropertyChecker.scala

```diff
@@ -0,0 +1,592 @@
+      } else {
+        (BooleanNode.getFalse, Array[String]("invalid_value"), typeString)
+      }
+    }
+  }
+
+  def stringProperty(typeString: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+    return (value, baseUrl, lang) => {
+      if (value.isTextual) {
+         (value, Array[String](), typeString)
+      } else {
+         (new TextNode(""), Array[String]("invalid_value"), typeString)
+      }
+    }
+  }
+
+  def numericProperty(typeString: PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+     return (value, baseUrl, lang) => {
+      if (value.isInt && value.asInt >= 0) {
+         (value, Array[String](), typeString)
+      } else if (value.isInt && value.asInt < 0) {
+         (NullNode.getInstance(), Array[String]("invalid_value"), typeString)
+      } else {
+         (NullNode.getInstance(), Array[String]("invalid_value"), typeString)
+      }
+    }
+  }
+
+  def notesProperty(typeString:PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+     def notesPropertyInternal (value: JsonNode, baseUrl: String, lang: String): (JsonNode, Array[String], PropertyType.Value) = {
+      if (value.isArray) {
+        val arrayValue = value.asInstanceOf[ArrayNode]
+        val elements = arrayValue.elements().asScala
+        if (elements forall (_.isTextual())) {
+          /**
+           * [(1,2), (3,4), (5,6)] => ([1,3,5], [2,4,6])
+           */
+          val (values, warnings) = Array.from(elements.map(x => checkCommonPropertyValue(x, baseUrl, lang))).unzip
+          val arrayNode:ArrayNode = PropertyChecker.mapper.valueToTree(values)
+          return (arrayNode, warnings, typeString)
+        }
+      }
+       (BooleanNode.getFalse, Array[String]("invalid_value"), typeString)
+    }
+    return notesPropertyInternal
+  }
+
+  def nullProperty(typeString:PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+     (value, baseUrl, lang) => {
+      if (value.isTextual) {
+         (value, Array[String](), typeString)
+      } else if (value.isArray) {
+        var values = Array[String]()
+        var warnings = Array[String]()
+        for (x <- value.elements().asScala) {
+          x match {
+            case xs if xs.isTextual => values = values :+ xs.asText()
+            case _ => warnings = warnings :+ "invalid_value"
+          }
+        }
+        val arrayNode:ArrayNode = PropertyChecker.mapper.valueToTree(values)
+         (arrayNode, warnings, typeString)
+      } else {
+         (NullNode.getInstance(), Array[String]("invalid_value"), typeString)
+      }
+    }
+  }
+
+  def separatorProperty(typeString:PropertyType.Value):(JsonNode, String, String) => (JsonNode, Array[String], PropertyType.Value) = {
+     (value, baseUrl, lang) => {
+       value match {
+        case s if s.isTextual => (s, null, typeString)
```

Comment:
Yes we have tried this, but for some unknown reason it didn't work. Now there are no complaints while using this syntax (or may be there is a slight difference now ?) 

Changed code to this multi conditions in case